### DISCUSSION TITLE
example question

### DISCUSSION THREAD

**Discussion by Raj2032** [DioxusLabs/dioxus#3189] [Q&A]
In your example:

```rs
fn app() -> Element {
    let mut count = use_signal(|| 0);

    rsx! {
        h1 { "High-Five counter: {count}" }
        button { onclick: move |_| count += 1, "Up high!" }
        button { onclick: move |_| count -= 1, "Down low!" }
    }
}
```

Was wondering can the data type for `count` be in `u8` instead of using the function: `use_signal(|| 0);` which I assume would assign the variable with a non-primative data type?

**Comment by ealmloff** [comment]
If I understand the question, you are asking why the example can't just be this?
```rust
fn app() -> Element {
    let mut count = 0;

    rsx! {
        h1 { "High-Five counter: {count}" }
        button { onclick: move |_| count += 1, "Up high!" }
        button { onclick: move |_| count -= 1, "Down low!" }
    }
}
```

There are two main reasons:
1) The app function reruns, and we don't want the value of count to reset every time. We put it in a hook so it keeps the value of `count` from the last time we ran the component
2) We need count to be [reactive](https://docs.rs/dioxus/0.6.0-alpha.4/dioxus/prelude/struct.ReactiveContext.html). When you read `count` somewhere, we need to keep track of that and rerun that part of the code the next time count changes. That helps dioxus keep UI and state up to date. For example, if you hit the up high button `"High-Five counter: {count}"` needs to update when the count changes to 1. You can read more about state and reactivity here: https://dioxuslabs.com/learn/0.6/essentials/state

**Reply by Raj2032** [reply]
@ealmloff makes sense, thanks
